internal/resources/metal/gateway/resource.go

```diff
@@ -0,0 +1,205 @@
+package gateway
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	equinix_errors "github.com/equinix/terraform-provider-equinix/internal/errors"
+	"github.com/equinix/terraform-provider-equinix/internal/framework"
+	"github.com/hashicorp/terraform-plugin-framework-timeouts/resource/timeouts"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+	"github.com/packethost/packngo"
+)
+
+func NewResource() resource.Resource {
+	r := Resource{
+		BaseResource: framework.NewBaseResource(
+			framework.BaseResourceConfig{
+				Name: "equinix_metal_gateway",
+			},
+		),
+	}
+	r.SetDefaultDeleteTimeout(20 * time.Minute)
+
+	return &r
+}
+
+type Resource struct {
+	framework.BaseResource
+	framework.WithTimeouts
```

Comment:
Since framework forces us to be more explicit about which resources/datasources supports timeouts, we should take the opportunity to document timeouts for resources that need them.  (If there are resources/datasources where we don't need them, we should remove the timeouts instead of documenting them, but it looks like this one was actually using the timeout block.)

Reply:
Updated: https://github.com/equinix/terraform-provider-equinix/pull/536/files#diff-438cdf7ccefabaebbc6e26c352807e69f690d3ea98273db2153ba7a7ce607cd1